options/add_musicalbum.rb

```diff
@@ -0,0 +1,25 @@
+require './classes/musicalbum'
+require './classes/genre'
+require './options/list_music_albums'
+
+def add_music_album
+  puts 'Enter album label:'
+  label = gets.chomp
+
+  puts 'Enter album genre:'
+  genre_name = gets.chomp
+  genre = @genres.find { |g| g.name == genre_name } || Genre.new(genre_name)
+  @genres << genre unless @genres.include?(genre)
+  puts 'Enter publish date (yyyy-mm-dd):'
+  publish_date = gets.chomp
+  puts 'Available on Spotify? (y/n):'
+  on_spotify = gets.chomp.downcase == 'y'
+
+  album = MusicAlbum.new(label, genre, publish_date, on_spotify, nil)
+  attribute_game_to_author(album)
+  @music_albums << album
+  genre.add_item(album)
+
+  puts 'Album added successfully!'
+  save_data
```

Comment:
I was thinking of saving the data of all files created in the session with one method. This is why in app.rb I have these lines:
```
if option.zero?
   save_all_files
   puts 'Thank you for using our app!'
```
It means whenever user types 0 - the app saves all files and exits the app. The implementation of save_all_files method is in preserve_data file. It has  the list of methods:
![image](https://github.com/Zilola-Nazarova/ruby-capstone-project/assets/61951420/b2b3ed33-e71e-48ca-949d-7a2c3b7775f8)

There you can define your method (in another branch).
This means you don't need to save_data in this file. Please let me know if you want to suggest a different approach.